pkg/codec/renamer.go

```diff
@@ -0,0 +1,76 @@
+package codec
+
+import (
+	"reflect"
+	"unicode"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+var pkgPath = reflect.TypeOf(renamer{}).PkgPath()
+
+func NewRenamer(fields map[string]string) Modifier {
+	m := &renamer{
+		modifierBase: modifierBase[string]{
+			fields:           fields,
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+	}
+	m.modifyFieldForInput = func(field *reflect.StructField, _, newName string) error {
+		field.Name = newName
+		if unicode.IsLower(rune(field.Name[0])) {
+			field.PkgPath = pkgPath
+		}
+		return nil
+	}
+	return m
+}
+
+type renamer struct {
+	modifierBase[string]
+}
+
+func (r *renamer) TransformForOffChain(output any) (any, error) {
+	rOutput, err := renameTransform(r.onToOffChainType, reflect.ValueOf(output))
+	if err != nil {
+		return nil, err
+	}
+	return rOutput.Interface(), nil
+}
+
+func (r *renamer) TransformForOnChain(input any) (any, error) {
+	rOutput, err := renameTransform(r.offToOnChainType, reflect.ValueOf(input))
+	if err != nil {
+		return nil, err
+	}
+	return rOutput.Interface(), nil
+}
```

Comment:
Does the implementation of `TransformOffChain` and `TransformOnChain` just happen to be identical for the renamer, or is this always going to be the case?  Wondering if we really need two different transformer methods or if we could get by with just one?

Reply:
It's not identical, one uses  r.onToOffChainType the other uses r.offToOnChainType.  They just happen to be similar for renamed because in both cases it just **safely** _unsafe_ casts to the other type :)

Reply:
Nevermind, I think it's probably better as-is I guess even if they were combined we'd still at least have to pass a boolean for which direction which is a more awkward API, and we can just implement it under the hood with a single method as you've done here.

Reply:
Most make helper calls that take different parameters depending on which way they go.  The chain implementation that combines multiple modifiers reverses the order if you call TransformOnChain :).

Reply:
Yeah, it makes sense